### PR TITLE
[SR-3279] Renaming Task to Process

### PR DESCRIPTION
**Pull request by damuellen**
Simple search and replace

### CODE REVIEW

File: Foundation/Process.swift [hunk 1]
```diff
@@ -457,7 +457,7 @@ extension Task {
     }
 }
 
-public let NSTaskDidTerminateNotification: String = "NSTaskDidTerminateNotification"
+public let ProcessDidTerminateNotification: String = "ProcessDidTerminateNotification"
```
Comment (ikesyo) [inline Foundation/Process.swift]: There should be no `ProcessDidTerminateNotification` on Darwin: https://developer.apple.com/reference/foundation/process/1413681-didterminatenotification

File: Foundation/Process.swift [hunk 2]
```diff
@@ -271,31 +271,31 @@ open class Task: NSObject {
...
-                    task.terminationHandler!(task)
+                    process.terminationHandler!(process)
                 }
                 thread.start()
             }
             
             // Set the running flag to false
             
-            task.running = false
+            process.running = false
```
Comment (ikesyo) [inline Foundation/Process.swift]: The property should be renamed to `isRunning` too: https://developer.apple.com/reference/foundation/process/1415788-isrunning

**Comment by damuellen** [comment]
Thanks @ikesyo
I have adapted it according to your comments.

File: Foundation/Process.swift [hunk 3]
```diff
@@ -422,42 +422,42 @@ open class Task: NSObject {
...
             
-        } while( self.running == true && RunLoop.current.run(mode: .defaultRunLoopMode, before: Date(timeIntervalSinceNow: 0.05)) )
+        } while( self.isRunning == true && RunLoop.current.run(mode: .defaultRunLoopMode, before: Date(timeIntervalSinceNow: 0.05)) )
         
         self.runLoop = nil
     }
 }
 
-public let NSTaskDidTerminateNotification: String = "NSTaskDidTerminateNotification"
+public let didTerminateNotification: String = "NSTaskDidTerminateNotification"
```
Comment (ikesyo) [inline Foundation/Process.swift]: This is not correct, the value is a static property and the type is `NSNotification.Name`:

```
extension Process {
    public static let didTerminateNotification: NSNotification.Name = ...
}
```

File: Foundation/Process.swift [hunk 4]
```diff
@@ -422,42 +422,42 @@ open class Task: NSObject {
...
-extension Task {
+extension Process {
     
     // convenience; create and launch
-    open class func launchedTaskWithLaunchPath(_ path: String, arguments: [String]) -> Task {
-        let task = Task()
-        task.launchPath = path
-        task.arguments = arguments
-        task.launch()
+    open class func launchedTaskWithLaunchPath(_ path: String, arguments: [String]) -> Process {
```
Comment (ikesyo) [inline Foundation/Process.swift]: This should be https://developer.apple.com/reference/foundation/process/1414375-launchedprocess

**Comment by ikesyo** [comment]
The type of `standardInput`, `standardOutput` and `standardError` should be changed to `Any?`.

File: Foundation.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -383,7 +383,7 @@
 		EADE0BB51BD15E0000C49C64 /* NSScanner.swift in Sources */ = {isa = PBXBuildFile; fileRef = EADE0B771BD15DFF00C49C64 /* NSScanner.swift */; };
 		EADE0BB61BD15E0000C49C64 /* NSSortDescriptor.swift in Sources */ = {isa = PBXBuildFile; fileRef = EADE0B781BD15DFF00C49C64 /* NSSortDescriptor.swift */; };
 		EADE0BB71BD15E0000C49C64 /* NSStream.swift in Sources */ = {isa = PBXBuildFile; fileRef = EADE0B791BD15DFF00C49C64 /* NSStream.swift */; };
-		EADE0BB81BD15E0000C49C64 /* NSTask.swift in Sources */ = {isa = PBXBuildFile; fileRef = EADE0B7A1BD15DFF00C49C64 /* NSTask.swift */; };
+		EADE0BB81BD15E0000C49C64 /* Process.swift in Sources */ = {isa = PBXBuildFile; fileRef = EADE0B7A1BD15DFF00C49C64 /* Process.swift */; };
```
Comment (phausler) [inline Foundation.xcodeproj/project.pbxproj]: I did not see a build.py alteration for this; might be a problem on linux builds (which is a pre-requisite to merging)

**Comment by phausler** [comment]
@swift-ci Please test

**Comment by damuellen** [comment]
Updated build.py and comments implemented

File: Foundation/Process.swift [hunk 6]
```diff
@@ -422,42 +422,42 @@ open class Task: NSObject {
...
             
-        } while( self.running == true && RunLoop.current.run(mode: .defaultRunLoopMode, before: Date(timeIntervalSinceNow: 0.05)) )
+        } while( self.isRunning == true && RunLoop.current.run(mode: .defaultRunLoopMode, before: Date(timeIntervalSinceNow: 0.05)) )
         
         self.runLoop = nil
     }
 }
 
-public let NSTaskDidTerminateNotification: String = "NSTaskDidTerminateNotification"
+public let didTerminateNotification = NSNotification.Name(rawValue: "NSTaskDidTerminateNotification")
```
Comment (ikesyo) [inline Foundation/Process.swift]: This is still defined as a global constant, not as a class property of `Process`.

**Comment by damuellen** [comment]
@ikesyo Thanks again for reviewing.

**Comment by parkera** [comment]
@swift-ci please test

**Pull request merged**